backend/src/index.ts

```diff
@@ -191,6 +209,55 @@ app.post("/login", async (req: TypedRequest<LoginBody>, res: Response) => {
     },
   );
 });
+// curl -H 'Content-Type: application/json' -d '{ "email": "ben", "color": "pink"}' -X POST http://localhost:3000/subscribe
+
+app.get(
+  "/startGame",
+  async (
+    req: TypedRequestQuery<{ roundCount: number; gameMode: Gamemode }>,
+    res,
+  ) => {
+    const getDoc = await getDocs(collection(db, "levels"));
+
+    const { roundCount, gameMode } = req.query;
+
+    // array of level IDs
+    const docIds = getDoc.docs.map(doc => doc.id);
+    const shuffled = docIds.sort(() => 0.5 - Math.random());
+
+    // Get sub-array of first n elements after shuffled
+    let selected = shuffled.slice(0, roundCount);
+    const levels: Level["id"][] = [];
+    selected.forEach((location) => levels.push(location))
+   
+    res.status(200).json(levels);
+  },
+);
+
+app.post(
+  "endGame", 
+  async (
+  req: TypedRequest<{gameMode: Gamemode, levels: Level["id"][], score: number}>, 
+  res: Response
+  ) => {
+    const { gameMode, levels, score } = req.body;
+  
+    const userId = await sessionIdToUserId(req.sessionID);
+
+      if (userId !== "guest") {
```

Comment:
i don't think this should be indented - run thru prettier

Reply:
i'd also recommend changing to something like
`if (userId === guest) ...` and return early